### PR TITLE
[CUDA graphs] [JIT] improves readability and nvfuser convenience for graph-safe cuda RNG

### PR DESCRIPTION
**Pull request by mcarilli**
I'm trying to make jitted RNG graph-safe in @csarofeen 's nvfuser branch. Doing so requires diffs in files outside torch/csrc/jit, and we'd like these to go upstream through the present simple separate PR (instead of needing to be reviewed as part of Christian's branch's eventual merge, which will be massive).

From the perspective of eager mode consumers, diffs here are purely cosmetic. I moved raw definitions of `PhiloxCudaState` and `at::cuda::philox::unpack` to standalone headers the codegen can easily copy from.

### CODE REVIEW

File: aten/src/ATen/CUDAGeneratorImpl.h [hunk 1]
```diff
@@ -56,7 +62,7 @@ namespace at {
  *
  * Example (see e.g. native/cuda/Dropout.cu):
  *
- * #include <ATen/CUDAGeneratorImpl.h>
+ * #include <ATen/cuda/CUDAGeneratorImpl.h>
```
Comment (ngimel) [inline aten/src/ATen/CUDAGeneratorImpl.h:65]: please undo this change, the correct path is ATen/CUDAGeneratorImpl.h

Comment (mcarilli) [reply aten/src/ATen/CUDAGeneratorImpl.h:65]: oops
